src/components/d-pad/d-pad.jsx

```diff
@@ -5,6 +5,16 @@ module.exports = React.createClass({
     onDirectionClick: React.PropTypes.func, // External handler for button clicks
     isVisible: React.PropTypes.bool
   },
+  // Do CSS updates that are programatically derived from the result of `getBoundingClientRect`
+  setDisplayState: function (boundingRect) {
+    var verticalMargin = boundingRect.top;
+    var horizontalMargin = boundingRect.left;
+
+    this.setState({
+      constrainedY: verticalMargin > 20 ? false : true,
```

Comment:
probably a good idea to store this as a constant somewhere. Also i'm feeling like it's a bit tight atm, might be a good idea to get a designer's eyes on it after though